connectors/byoc.py

```diff
@@ -650,6 +681,55 @@ async def document_count(self):
         )
         return result["count"]
 
+    async def check_configuration_formatting(self, fqn, service_type):
+        try:
+            source_klass = get_source_klass(fqn)
+        except Exception as e:
+            logger.critical(e, exc_info=True)
+            raise DataSourceError(f"Could not instantiate {fqn} for {service_type}")
+
+        simple_config = source_klass.get_simple_configuration()
+        current_config = self.configuration.to_dict()
+
+        # raise an error if the configuration is missing any fields
+        missing_fields = list(set(simple_config.keys()) - set(current_config.keys()))
+
+        if len(missing_fields) > 0:
+            raise MalformedConfigurationError(
+                f'Connector for {service_type}(id: "{self.id}") has missing configuration fields: {", ".join(missing_fields)}'
+            )
+
+        # if a field is missing properties, add them with default values
+        configs_missing_properties = filter_nested_dict_by_keys(
+            DEFAULT_CONFIGURATION.keys(), current_config
+        )
+        if not configs_missing_properties:
+            return
+
+        logger.debug(
```

Comment:
I would make it more verbose. I often run the service without `--debug` flag so I will miss this.

Reply:
I've changed this logger to be `.info`. Do you think more logs could be added, or just this one more visible?

Reply:
it looks good to me